pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -17,44 +22,48 @@
 , url
 , hash ? null
 , sha256 ? null
-, supportedCudaVersions ? []
+, supportedCudaVersions ? [ ]
 }:
 
 assert (hash != null) || (sha256 != null);
 
 let
+  inherit (cudatoolkit) cc;
+
   majorMinorPatch = version: lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
   version = majorMinorPatch fullVersion;
-in stdenv.mkDerivation {
-  name = "cudatoolkit-${cudatoolkit.majorVersion}-cudnn-${version}";
+in
+stdenv.mkDerivation {
+  name = "cudatoolkit-${cudaMajorVersion}-cudnn-${version}";
 
   inherit version;
-  # It's often the case that the src depends on the version of cudatoolkit it's
-  # being linked against, so we pass in `cudatoolkit` as an argument to `mkSrc`.
   src = fetchurl {
     inherit url hash sha256;
   };
 
-  nativeBuildInputs = [ addOpenGLRunpath ];
+  # Check and normalize Runpath against DT_NEEDED using autoPatchelf.
+  # Prepend /run/opengl-driver/lib using addOpenGLRunpath
+  # so that libcuda (which is not part of DT_NEEDED)
+  # can be found at runtime with dlopen().
+  nativeBuildInputs = [
+    autoPatchelfHook
+    autoAddOpenGLRunpathHook
+  ];
+
+  # Used by autoPatchelfHook
+  buildInputs = builtins.map lib.getLib [
+    cc.cc # libstdc++
+    libcublas
+    zlib
+  ];
```

Comment:
cudatoolkit is still needed for versions where redist packages are not available

Reply:
Are these versions still being supported once 22.05 is released?

Reply:
@FRidh were you meaning to backport the zlib fix? Or is it that you want to keep the function's interface compatible with hypothetical out-of-tree overrides?

Reply:
I'd much rather add a record in release notes for 22.05 and keep things clean. Those who have out-of-tree overrides would get "called with unexpected arguments" error once and update

Reply:
> Are these versions still being supported once 22.05 is released? 

We still package them so I suppose so?

Reply:
> We still package them so I suppose so?

/* prolonged miscommunication vibes */ I repeated the word "versions", but I'm really not sure what you were referring to. Versions of what?

Reply:
Aghh 🤦🏻 you mean  cudatolkit<11